pkg/controllers/rollout/controller_test.go

```diff
@@ -155,59 +158,380 @@ func TestReconcilerHandleResourceSnapshot(t *testing.T) {
+						placementv1beta1.PlacementTrackingLabel: "placement",
+					},
+				},
+			},
+			wantEnqueueKeys: []reconcile.Request{
+				{NamespacedName: types.NamespacedName{Name: "placement", Namespace: "test-namespace"}},
+			},
 		},
 		"test skip a malformatted resourceBinding": {
 			resourceBinding: &placementv1beta1.ClusterResourceSnapshot{
 				ObjectMeta: metav1.ObjectMeta{},
 			},
-			shouldEnqueue: false,
+			wantEnqueueKeys: nil,
 		},
 	}
 	for name, tt := range tests {
 		t.Run(name, func(t *testing.T) {
 			queue := &controllertest.Queue{TypedInterface: workqueue.NewTypedRateLimitingQueue[reconcile.Request](workqueue.DefaultTypedItemBasedRateLimiter[reconcile.Request]())}
-			enqueueResourceBinding(tt.resourceBinding, queue)
-			if tt.shouldEnqueue && queue.Len() == 0 {
-				t.Errorf("enqueueResourceBinding test `%s` didn't queue the object when it should enqueue", name)
-			}
-			if !tt.shouldEnqueue && queue.Len() != 0 {
-				t.Errorf("handleResourceSnapshot test `%s` queue the object when it should not enqueue", name)
-			}
+			enqueueBinding(tt.resourceBinding, queue)
+			validateEnqueueBehavior(t, name, queue, tt.wantEnqueueKeys)
+		})
+	}
+}
+
+func TestReconcilerHandleResourceOverrideSnapshot(t *testing.T) {
+	tests := map[string]struct {
+		snapshot        client.Object
+		enqueueCRP      bool
+		wantEnqueueKeys []reconcile.Request
+	}{
+		"test enqueue a new latest ResourceOverrideSnapshot with cluster scoped placement": {
+			snapshot: &placementv1beta1.ResourceOverrideSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "override-snapshot-1",
+					Namespace: "test-namespace",
+					Labels: map[string]string{
+						placementv1beta1.IsLatestSnapshotLabel: "true",
+					},
+				},
+				Spec: placementv1beta1.ResourceOverrideSnapshotSpec{
+					OverrideSpec: placementv1beta1.ResourceOverrideSpec{
+						Placement: &placementv1beta1.PlacementRef{
+							Name:  "test-placement",
+							Scope: placementv1beta1.ClusterScoped,
+						},
+					},
+				},
+			},
+			enqueueCRP: true,
+			wantEnqueueKeys: []reconcile.Request{
+				{NamespacedName: types.NamespacedName{Name: "test-placement"}},
+			},
+		},
+		"test skip a new latest ResourceOverrideSnapshot with cluster scoped placement while enqueueCRP=false": {
+			snapshot: &placementv1beta1.ResourceOverrideSnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "override-snapshot-1",
+					Namespace: "test-namespace",
+					Labels: map[string]string{
+						placementv1beta1.IsLatestSnapshotLabel: "true",
+					},
+				},
+				Spec: placementv1beta1.ResourceOverrideSnapshotSpec{
+					OverrideSpec: placementv1beta1.ResourceOverrideSpec{
+						Placement: &placementv1beta1.PlacementRef{
+							Name:  "test-placement",
+							Scope: placementv1beta1.ClusterScoped,
+						},
+					},
+				},
+			},
+			enqueueCRP:      false,
+			wantEnqueueKeys: nil,
+		},
+		"test enqueue a new latest ResourceOverrideSnapshot with namespace scoped placement": {
```

Comment:
```suggestion
		"test enqueue a new latest ResourceOverrideSnapshot with namespace scoped placement while enqueueCRP=false": {
```

the test name is not consistent with the above one